pkg/apis/options/legacy_options_test.go

```diff
@@ -759,4 +763,87 @@ var _ = Describe("Legacy Options", func() {
 			}),
 		)
 	})
+
+	Context("Legacy Providers", func() {
+		type convertProvidersTableInput struct {
+			legacyProvider    LegacyProvider
+			expectedProviders Providers
+			errMsg            string
+		}
+
+		// Non defaults for these options
+		clientID := "abcd"
+
+		defaultProvider := Provider{
+			ProviderID:   "google_" + clientID,
+			ClientID:     clientID,
+			ProviderType: "google",
+		}
+		defaultLegacyProvider := LegacyProvider{
+			ClientID:     clientID,
+			ProviderType: "google",
+		}
+
+		displayNameProvider := Provider{
+			ProviderID:   "displayName",
+			ProviderName: "displayName",
+			ClientID:     clientID,
+			ProviderType: "google",
+		}
+
+		displayNameLegacyProvider := LegacyProvider{
+			ClientID:     clientID,
+			ProviderName: "displayName",
+			ProviderType: "google",
+		}
+
+		internalConfigProvider := Provider{
+			ProviderID:   "google_" + clientID,
+			ClientID:     clientID,
+			ProviderType: "google",
+			GoogleConfig: GoogleOptions{
+				GoogleAdminEmail:         "email@email.com",
+				GoogleServiceAccountJSON: "test.json",
+				GoogleGroups:             []string{"1", "2"},
+			},
+		}
+
+		internalConfigLegacyProvider := LegacyProvider{
+			ClientID:                 clientID,
+			ProviderType:             "google",
+			GoogleAdminEmail:         "email@email.com",
+			GoogleServiceAccountJSON: "test.json",
+			GoogleGroups:             []string{"1", "2"},
+		}
+		DescribeTable("convertLegacyProviders",
+			func(o *convertProvidersTableInput) {
+
+				providers, err := o.legacyProvider.convert()
+
```

Comment:
Nit: We don't need these new lines really
```suggestion
				providers, err := o.legacyProvider.convert()
```